sympy/matrices/sparse.py

```diff
@@ -985,8 +985,10 @@ def col_join(self, other):
         >>> C == A.row_insert(A.rows, Matrix(B))
         True
         """
-        if not self:
-            return type(self)(other)
+        # A null matrix can always be stacked (see  #10770)
+        if self.rows == 0 and self.cols != other.cols:
```

Comment:
If they're equal it will pass through and the rest of the routine will handle it correctly.  This is to catch the special case from #10770 where a 0x anything matrix can be stacked with another even if the sizes are incompatible.